hack/prefetch-images.sh

```diff
@@ -0,0 +1,61 @@
+set -ex pipefail
+
+source hack/common.sh
+source cluster-up/cluster/$KUBEVIRT_PROVIDER/provider.sh
+source hack/config.sh
+
+function prefetch-images::find_node_names() {
+    if [[ $KUBEVIRT_PROVIDER == "external" ]] || [[ $KUBEVIRT_PROVIDER =~ kind.* ]] || [[ $KUBEVIRT_PROVIDER == "local" ]]; then
+        echo "" # in case of external provider / kind we have no control over the nodes
+    else
+        local nodes=()
+        nodes+=($(_kubectl get nodes -o name | sed "s#node/##g"))
+        echo "${nodes[@]}"
+    fi
+}
+
+function prefetch-images::find_pull_command() {
+    if [[ $KUBEVIRT_PROVIDER == "external" ]] || [[ $KUBEVIRT_PROVIDER =~ kind.* ]] || [[ $KUBEVIRT_PROVIDER == "local" ]]; then
+        echo "" # in case of external provider / kind we have no control over the nodes
```

Comment:
If this returns empty, it will blow up at the caller when attempting to execute. Why do we need it at all?
Looks to me like the `find_node_names` is supposed to protect against using it.

Reply:
Do you have any idea/suggestion how to write it better, so it is ready for the introduction of podman/other tools, or maybe to just simplify this?

Bash for me is the tool to write a 10 line scripts, I am not familiar with clean code/easy to maintain bash code idioms so any help is appreciated.

Reply:
Lets try to examine just the logic: Why do we need this function at all? We will not pull for these providers because the nodes are not collected.

Do I miss something?